examples/image-resize/README.md

```diff
@@ -0,0 +1,45 @@
+# Image Resize
+
+It uses the excellent [https://sharp.pixelplumbing.com/](sharp) npm package
+
+## Preview
+
+Open this example on [CodeSandbox](https://codesandbox.com):
+
+[![Open in CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/remix-run/remix/tree/main/examples/image-resize)
+
+## Example
+
+From your terminal:
+
+```sh
+npm run dev
+```
+
+Visit
+http://localhost:3000/ to view the resized images
+
+The `Image` component is a simple wrapper which creates the resize url for our `/assets/resize/$.ts` route
+
+```
+http://localhost:3000/assets/resize/dog-1.jpg?w=600&h=600&fit=cover
```

Comment:
Could you make this work to point to external images as well?

```suggestion
http://localhost:3000/assets/resize?url=%2Fdog-1.jpg&w=600&h=600&fit=cover
```

Then you could optimize even hot-linked images 🔥

Reply:
We could argue that that's beyond the scope of this example, but maybe you could add a comment about this and say it's possible and leave it as an exercise for the reader.

Reply:
This isn't really a scalable demo, since usually you don't store your images in the source code, if it isn't your personal blog.

The code for creating a read stream to GCP isn't much

```
const {Storage} = require('@google-cloud/storage');

const storage = new Storage();

// create read stream directly to gcp bucket
const bucketName = 'my-gcp-bucket'
const bucketPath = '/dogs/cute/dog-1.jpg'
const readStream = storage.bucket(bucketName).file(bucketPath).createReadStream()

```

but since it requires that we have a gcp bucket and permissions to read from it, it would require some more work.
But I think it will be something people will ask for.

Should we add it as a secondary route, `/assets/google-storage-resize/$.ts`?

or maybe a totally different example?

Reply:
I added a commented out section for both GCP and S3